src/util.c

```diff
@@ -493,6 +621,17 @@ int string2ll(const char *s, size_t slen, long long *value) {
     return 1;
 }
 
+int string2ll(const char *s, size_t slen, long long *value) {
+#if HAVE_AVX512
+    /* TODO: Cache the cpu info when server startup */
+    if (__builtin_cpu_supports("avx512f") &&
+        __builtin_cpu_supports("avx512vl") &&
+        __builtin_cpu_supports("avx512bw"))
```

Comment:
Should we fix the TODO? If it doesn't matter, then we can remove the TODO comment.

Reply:
I just remove the TODO comments; I will send another PR to address that.

Reply:
Ah, if you want to fix it in the future, it's OK to keep the TODO comment as a reminder. I just assumed we will never fix it and it's not important.

Can we use a simple static variable?

```c
    static int cpu_supports_avx512 = __builtin_cpu_supports("avx512f") && ...;
    if (cpu_supports_avx512) ...
```

Reply:
> Can we use a simple static variable?
> 
> ```c
>     static int cpu_supports_avx512 = __builtin_cpu_supports("avx512f") && ...;
>     if (cpu_supports_avx512) ...
> ```

The `__builtin_cpu_supports()` function is meant to be executed at runtime to determine CPU capabilities. When used in a static initializer context like this, it evaluates at compile time based on the compilation target rather than the actual CPU where the code will run.

Reply:
I prefer to retrieve the CPUID information during server startup. We could introduce a new server data structure, `cpuid`, to check for support of SSE, SSE2, AVX2, AVX512, AVX10, and other SIMD instructions.

Reply:
If it is, then why isn't also every `__builtin_cpu_supports()` evaluated at compile time even in code like ↓ ?

```C
    if (__builtin_cpu_supports("avx512f") &&
        __builtin_cpu_supports("avx512vl") &&
        __builtin_cpu_supports("avx512bw"))
```

Reply:
When I compile code like ```static int cpu_supports_avx512 = __builtin_cpu_supports("avx512f");```
error: initializer element is not constant.